src/Draco.Compiler/Internal/Documentation/SymbolDocumentation.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Xml.Linq;
+
+namespace Draco.Compiler.Internal.Documentation;
+
+/// <summary>
+/// Represents documentation for a <see cref="Symbols.Symbol"/>.
+/// </summary>
+/// <param name="Sections">The <see cref="DocumentationSection"/>s this documentation contains.</param>
+internal record class SymbolDocumentation(ImmutableArray<DocumentationSection> Sections)
```

Comment:
What if we only exposed ordered sections as `Sections`? We wouldn't have to have 2 props in the API. The sections passed in the ctor can be stored as a `private readonly T unorderedSections`